torch/cuda/amp/autocast_mode.py

```diff
@@ -0,0 +1,217 @@
+import torch
+import functools
+import warnings
+import numpy as np
```

Comment:
Does this introduce a new dependency on NumPy? Or is NumPy now already a dependency?

Earlier, when it was not a dependency, in collation function a workaround had been used: https://github.com/pytorch/pytorch/blob/master/torch/utils/data/_utils/collate.py#L23

Reply:
Doesn't Pytorch already depend on Numpy?  grepping for "import numpy" yields a lot of hits.

FYI these diffs were already approved and merged.  They were reverted because they broke a windows build for an old version of Visual Studio that wasn't represented in public CI.  This resubmission adds minor changes to address the build failure.

Reply:
If I understand well, PyTorch originally did not depend in NumPy. Most of occurences of "import numpy" were either tests or conditional imports.

However, this situation seems to have changed yesterday :) https://github.com/pytorch/pytorch/commit/275f5c80491d81f53677eaaf277aa875b094788e - yesterday's commit.

cc @ezyang

Reply:
i'll call that a rare stroke of luck

Reply:
:)